CSVZipExport/module.php

```diff
@@ -85,44 +85,75 @@ public function Export(int $ArchiveVariable, int $AggregationStage, string $Aggr
-                    $endTimeStamp = $this->TransferTime($AggregationEnd, false, false);
-                    break;
-                case 1: //Daily
-                case 2: //Weekly
-                case 3: //Monthly
-                case 4: //Yearly
-                    $startTimeStamp = $this->TransferTime($AggregationStart, true, true);
-                    $endTimeStamp = $this->TransferTime($AggregationEnd, false, true);
-                    break;
-            }
+            case 0: //Hourly
+            case 5: //1-Minute
+            case 6: //5-Minute
+            case 7: //raw datas
+                $startTimeStamp = $this->TransferTime($AggregationStart, true, false);
+                $endTimeStamp = $this->TransferTime($AggregationEnd, false, false);
+                break;
+            case 1: //Daily
+            case 2: //Weekly
+            case 3: //Monthly
+            case 4: //Yearly
+                $startTimeStamp = $this->TransferTime($AggregationStart, true, true);
+                $endTimeStamp = $this->TransferTime($AggregationEnd, false, true);
+                break;
+        }
 
-        //Generate zip with aggregated values
-        $tempfile = sys_get_temp_dir() . DIRECTORY_SEPARATOR . $this->GenerateFileName($ArchiveVariable);
-        $zip = new ZipArchive();
-        if ($zip->open($tempfile, ZipArchive::CREATE | ZipArchive::OVERWRITE) === true) {
+        $contentFile = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'ContentTemp.txt';
+        file_put_contents($contentFile, ''); //Create the tempfile
+        $loopAgain = true;
+        $endElements = [];
+
+        while ($loopAgain) {
             $content = '';
             if ($AggregationStage != 7) {
-                $loggedValues = AC_GetAggregatedValues($archiveControlID, $ArchiveVariable, $AggregationStage, $startTimeStamp, $endTimeStamp, 0);
+                $loggedValues = AC_GetAggregatedValues($archiveControlID, $ArchiveVariable, $AggregationStage, $startTimeStamp, $endTimeStamp - 1, $limit);
+                $loopAgain = count($loggedValues) == $limit;
                 for ($j = 0; $j < count($loggedValues); $j++) {
                     $content .= date('d.m.Y H:i:s', $loggedValues[$j]['TimeStamp']) . ';' . $loggedValues[$j]['Avg'] . "\n";
                 }
             } else {
-                $loggedValues = AC_GetLoggedValues($archiveControlID, $ArchiveVariable, $startTimeStamp, $endTimeStamp, 0);
+                $loggedValues = AC_GetLoggedValues($archiveControlID, $ArchiveVariable, $startTimeStamp, $endTimeStamp, $limit);
+                $loopAgain = count($loggedValues) == $limit;
+
+                //Protect values to duplicate on limit border
+                foreach ($endElements as $element) {
+                    if ($element == reset($loggedValues)) {
+                        array_shift($loggedValues);
+                    }
+                }
```

Comment:
Funktioniert das? Du nutzt ja prinzipiell ==-Vergleiche auf assoziativen Arrays. Ich wusste nicht, dass das funktioniert, aber scheinbar schon. Aber scheitert das nicht, wenn mehrfach der gleiche Eintrag kommt? Also beispielsweise auf einem Zeitstempel von 1 auf 0 auf 1 auf 0 gewechselt wird? Ich glaube, wenn du jetzt nur die ersten 1 und 0 erwischst, wirst du beim nächsten Durchlauf auch den dritten und vierten Wert rausfiltern, oder?

Reply:
Mein Ansatz wäre hier übrigens ein anderer gewesen. Wenn noch ein Schleifendurchgang kommt, dann würde ich einfach alle Einträge mit dem ältesten Zeitstempel rauswerfen (oder gar nicht erst in die Liste packen), du fragst sie im nächsten Durchlauf ja sowieso nochmal mit ab.